timemachine/cpp/src/gpu_utils.cuh

```diff
@@ -34,10 +34,22 @@ inline void curandAssert(curandStatus_t code, const char *file, int line, bool a
     }
 }
 
+/* cudaSafeMalloc is equivalent to gpuErrchk(cudaMalloc(...)) except that it prints a warning message if the
+* allocation is greater than a GiB.
+*/
+#define cudaSafeMalloc(ptr, size)                                                                                      \
```

Comment:
I tried to follow the previous style of inlining a function in the macro. However it needs to handle all of the different types. Avoided adding templating, but could do that if it is preferred.